test/kafka_connect_sink_crud_tests/test_crud.py

```diff
@@ -11,169 +11,182 @@
-               "splunk.hec.raw": "disable",
-               "splunk.hec.ack.enabled": "false",
-               "splunk.hec.ssl.validate.certs": "false"
-              }
-            }
-
-    assert create_kafka_connector(setup, connector_definition_invalid_tasks) == expected
-
-    # connector definition with topics invalid(empty string)
-    connector_definition_invalid_tasks = {
-            "name": "kafka-connect-splunk",
-            "config": {
-               "connector.class": "com.splunk.kafka.connect.SplunkSinkConnector",
-               "tasks.max": "3",
-               "topics": "",
-               "splunk.indexes": setup["splunk_index"],
-               "splunk.hec.uri": setup["splunkd_url"],
-               "splunk.hec.token": setup["splunk_token"],
-               "splunk.hec.raw": "false",
-               "splunk.hec.ack.enabled": "false",
-               "splunk.hec.ssl.validate.certs": "false"
-              }
-            }
-
-    assert create_kafka_connector(setup, connector_definition_invalid_tasks) == expected
-
-    # connector definition with splunk.hec.json.event.enrichment invalid(non key value pairs)
-    connector_definition_invalid_tasks = {
-            "name": "kafka-connect-splunk",
-            "config": {
-               "connector.class": "com.splunk.kafka.connect.SplunkSinkConnector",
-               "tasks.max": "3",
-               "topics": setup["kafka_topic"],
-               "splunk.indexes": setup["splunk_index"],
-               "splunk.hec.uri": setup["splunkd_url"],
-               "splunk.hec.token": setup["splunk_token"],
-               "splunk.hec.raw": "false",
-               "splunk.hec.ack.enabled": "false",
-               "splunk.hec.ssl.validate.certs": "false",
-               "splunk.hec.json.event.enrichment": "testing-testing non KV"
-              }
-            }
-
-    assert create_kafka_connector(setup, connector_definition_invalid_tasks) == expected
-
-    # connector definition with splunk.hec.json.event.enrichment invalid(key value pairs not separated by commas)
-    connector_definition_invalid_tasks = {
-            "name": "kafka-connect-splunk",
-            "config": {
-               "connector.class": "com.splunk.kafka.connect.SplunkSinkConnector",
-               "tasks.max": "3",
-               "topics": setup["kafka_topic"],
-               "splunk.indexes": setup["splunk_index"],
-               "splunk.hec.uri": setup["splunkd_url"],
-               "splunk.hec.token": setup["splunk_token"],
-               "splunk.hec.raw": "false",
-               "splunk.hec.ack.enabled": "false",
-               "splunk.hec.ssl.validate.certs": "false",
-                "splunk.hec.json.event.enrichment": "key1=value1 key2=value2"
-              }
-            }
-
-    assert create_kafka_connector(setup, connector_definition_invalid_tasks) == expected
+
+class TestCrud:
+
+    @pytest.fixture(scope='class', autouse=True)
+    def setup_class(self, setup):
```

Comment:
why do we need this method if we have an unconfigure method?

Reply:
This method is class level fixture. I added `setup['connectors'] = []` to create a list for connectors that will be create in the test case, and after `yield` we do class level tear down, to remove which ever connectors created in this class but failed to remove. For example from line 53 to line 89, if any assertion failed in the middle, the connector will stay around and affect other tests result. The `unconfigure` method can only remove the connectors that created in the beginning of the session based on the data in `connect_params.py`, it will not delete the connector that created on line 53.